src/client/contextChannels.ts

```diff
@@ -2,79 +2,424 @@
+ * @param channelId The ID of the channel to fetch.
+ */
+export async function getChannelById(channelId: ChannelId): Promise<Channel|null> {
+    return channelLookup[channelId] || null;
+}
+
+/**
+ * Gets all user-visible channels.
+ *
+ * This is the list of channels that should be used to populate a channel selector. All channels returned will have
+ * additional metadata that can be used to populate a selector UI with a consistent cross-app channel list.
+ */
+export async function getDesktopChannels(): Promise<DesktopChannel[]> {
+    const channelData: ChannelTransport[] = await tryServiceDispatch(APIFromClientTopic.GET_DESKTOP_CHANNELS, {});
+
+    return channelData.map(getChannelObject) as DesktopChannel[];
+}
+
+/**
+ * Returns the channel that the current window is assigned to.
+ *
+ * @param identity The window to query. If omitted, will use the window that calls this method.
+ * @throws `FDC3Error`: If `identity` is not a valid {@link https://developer.openfin.co/docs/javascript/stable/global.html#Identity | Identity}
+ * @throws `FDC3Error`: If the window specified by `identity` does not exist
+ * @throws `FDC3Error`: If the window specified by `identity` does not integrate FDC3 (determined by inclusion of the client API module)
+ */
+export async function getCurrentChannel(identity?: Identity): Promise<Channel> {
+    const channelData: ChannelTransport = await tryServiceDispatch(APIFromClientTopic.GET_CURRENT_CHANNEL, {identity});
+
+    return getChannelObject(channelData);
+}
+
+/**
+ * Event that is fired whenever a window changes from one channel to another.
+ *
+ * Note that this listener applies to all windows and channels. Check the properties on the event to see which window
+ * and channels were modified.
  */
-export async function getChannelMembers(id: ChannelId): Promise<Identity[]> {
-    return tryServiceDispatch(APIFromClientTopic.GET_CHANNEL_MEMBERS, {id});
+export function addEventListener(eventType: 'channel-changed', handler: (event: ChannelChangedEvent) => void): void;
+
+export function addEventListener<E extends ChannelEvent>(eventType: E['type'], handler: (event: E) => void): void {
+    eventEmitter.addListener(eventType, handler);
+}
+
+export function removeEventListener(eventType: 'channel-changed', handler: (event: ChannelChangedEvent) => void): void;
+
+/**
+ * Removes a listener previously added with {@link addEventListener}.
+ *
+ * Has no effect if `eventType` isn't a valid event, or `listener` isn't a callback registered against `eventType`.
+ *
+ * @param eventType The event being unsubscribed from
+ * @param listener The callback function to remove
+ */
+export function removeEventListener<E extends ChannelEvent>(eventType: E['type'], handler: (eventPayload: E) => void): void {
+    eventEmitter.removeListener(eventType, handler);
+}
+
+const channelLookup: {[id: string]: Channel} = {
+    [DEFAULT_CHANNEL_ID]: defaultChannel
```

Comment:
Pretty arbitrary decision really. Mainly to keep things "lightweight", since we only need a string key. Will also never be removing or filtering, or anything like that. The `Map` type is mostly useful when you want keys that are types other than string.